ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedRowBatchCtx.java

```diff
@@ -476,8 +473,8 @@ public void addPartitionColsToBatch(VectorizedRowBatch batch, Object[] partition
             bcv.isNull[0] = true;
             bcv.isRepeating = true;
           } else {
-            bcv.fill(sVal.getBytes());
-            bcv.isNull[0] = false;
+            bcv.setVal(0, sVal.getBytes());
```

Comment:
I figured out what was causing the NPE. HIVE-9937 (committed in Hive 1.2.0) added a line `vc.init();` to `VectorizedRowBatch.reset()`, which was needed.